src/main/java/com/carservice/manager/repository/VehicleRepository.java

```diff
@@ -20,48 +19,44 @@
 
     private VehicleModel vehicleModel;
 
-    public List<VehicleModel> getVehicleModels() {
-        return readDataFromFile();
+    public List<VehicleModel> getVehicleModels(String fileName) {
+        return readDataFromFile(fileName);
     }
 
 
-    public void addVehicle(VehicleModel vehicleModel) {
-        List<VehicleModel> vehicleModels = readDataFromFile();
+    public void addVehicle(VehicleModel vehicleModel, String fileName) {
+        List<VehicleModel> vehicleModels = readDataFromFile(fileName);
         vehicleModels.add(vehicleModel);
-        saveJson(vehicleModels);
+        saveJson(vehicleModels, fileName);
 
     }
 
-    public void addAll(List<VehicleModel> vehicleModel) {
-        saveJson(vehicleModel);
+    public void addAll(List<VehicleModel> vehicleModel, String fileName) {
+        saveJson(vehicleModel, fileName);
     }
 
 
-    private void saveJson(Object object) {
+    private void saveJson(Object object, String fileName) {
         Gson gson = new Gson();
```

Comment:
Make gson as a class property, not a local variable, so one instance can be reused 😉